dev/bots/analyze_framework_code.dart

```diff
@@ -0,0 +1,421 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/analysis_context.dart';
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/analysis/session.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/source/line_info.dart';
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as path;
+
+import 'utils.dart';
+
+Future<void> runVerifiersInResolvedDirectory(String workingDirectory, List<ResolvedUnitVerifier> verifiers) async {
```

Comment:
workingDirectory == flutterRepoRoot? It has to be a specific working directory, right? Passing in anything as the working dir will not work...

Reply:
in `bots/analyze.dart` yeah but the function is also called in a test so it can't hard code the path.

Reply:
What I meant, let's give it a better name. It's not any working directory, it has to be a directory that contains the flutter root. It's still fine to have this as an argument and make it configureable where it is.